### DISCUSSION TITLE
Calculation of ODF based on pole figure export data

### DISCUSSION THREAD

**Discussion by tarekiraki** [mtex-toolbox/mtex#1277] [Ask Anything] [answered]
Dear Sir or Madam,

I would like to calculate an ODF based on exported Pole Figure data. My source-code is listed below. The resulting figures are attached at the end.

For the ODF I used the example [PoleFigure2ODFAmbiguity](https://mtex-toolbox.github.io/PoleFigure2ODFAmbiguity.html), with the change that I only used 3 miller indices (step 1). It is visualized in Figure 1. Next, the pole figure was calculated based on this ODF (step 2). The ODF was recalculated based on this pole figure (step 3) and is visualized in Figure 2.

I would like to use this procedure on file level. For this purpose, the pole figure from (step 2) is exported [using the export function for pole figures](https://mtex-toolbox.github.io/PoleFigure.export.html) (step 4). This exports three .txt files, one file for each miller index. The pole figure is then imported (step 5), the ODF is calculated (step 6), and visualized in Figure 3.

Unfortunately, Figure 3 does not look anything like Figure 1 and Figure 2.
So I would like to ask why that could be. Is there anything else to consider when importing and exporting to the file system?

```
% Source: https://mtex-toolbox.github.io/PoleFigure2ODFAmbiguity.html

clear all; close all, clc;

% import mtex
addpath('/home/tarek/Software/mtex-5.7.0')
startup_mtex;

% define crystalSymmetry and Miller. use only three miller indices.
cs = crystalSymmetry('mmm');
%h = Miller({1,0,0},{0,1,0},{0,0,1},{1,1,0},{1,0,1},{0,1,1},{1,1,1},{1,2,0},cs);
h = Miller({1,0,0},{0,1,0},{0,0,1},cs);

% step 1) define ODF original
orix = orientation.byAxisAngle(xvector,90*degree,cs);
oriy = orientation.byAxisAngle(yvector,90*degree,cs);
oriz = orientation.byAxisAngle(zvector,90*degree,cs);
odf_orig = unimodalODF([orix,oriy,oriz]);

% figure 1: ODF original
figure(1)
plotPDF(odf_orig, h, 'contourf')

% step 2) calculate Pole Figure from ODF original
pf_odf_orig = calcPoleFigure(odf_orig, h, 'upper');

% step 3) calculate ODF based Pole Figure 
odf1_recon = calcODF(pf_odf_orig,'silent');

% figure 2: ODF based on Pole Figure 
figure(2)
plotPDF(odf1_recon,h,'contourf')

% step 4) export Pole Figure to file system
folder_pf = 'insertfolderhere';
export(pf_odf_orig,folder_pf)   
% this gives three data files.

% step 5) import Pole Figure data
fnames = {
    fullfile(folder_pf, '/_(100).txt'), 
    fullfile(folder_pf, '/_(010).txt'),
    fullfile(folder_pf, '/_(001).txt')
    };
pf_import = loadPoleFigure(fnames, cs, h, 'interface','generic', 'ColumnNames', {'Polar Angle' 'Azimuth Angle' 'Intensity'});

% step 6) calulate ODF based on Pole Figure data
odf_recon = calcODF(pf_import,'silent');

% figure 3): ODf based on Pole Figure data from file system.
figure(3)
plotPDF(odf_recon, h,'contourf')
```

![Figure1](https://user-images.githubusercontent.com/5840093/141101115-2e3e70cc-f135-4c93-81ac-6e3b40f899a6.png)
![Figure2](https://user-images.githubusercontent.com/5840093/141101140-0bc64c61-d461-4dd7-848b-51b6637b9d29.png)
![Figure3](https://user-images.githubusercontent.com/5840093/141101160-fc79afb0-f40b-4e6e-a7ad-c82b5a1bd337.png)

**Comment by kilir** [comment, marked as answer]
Hi,
two comments, you export in radians, hence the import needs to know that
```
pf_import = PoleFigure.load(fnames, cs, h, 'ColumnNames', {'Polar Angle' 'Azimuth Angle' 'Intensity'},'RADIANS');
```
and while not a thing in your example, it might be more useful to use {001} {110} and e.g. {111} instead of three times the same polefigure.

Cheers,
Rüdiger

**Comment by tarekiraki** [comment]
Dear Rüdiger,

thank you very much for your comments. Now it works, as shown in the Figures.

Cheers, 
Tarek

![Fig1](https://user-images.githubusercontent.com/5840093/141125586-e1715756-3080-4bd4-97f1-504a60972062.png)
![Fig2](https://user-images.githubusercontent.com/5840093/141125661-8b94315b-e94f-4d9d-9124-f93afcc03a26.png)
![Fig3](https://user-images.githubusercontent.com/5840093/141125679-8159ea5b-ece4-4b16-800e-b67583d47a83.png)
